holidays/countries/tajikistan.py

```diff
@@ -0,0 +1,116 @@
+#  holidays
+#  --------
+#  A fast, efficient Python library for generating country, province and state
+#  specific sets of holidays on the fly. It aims to make determining whether a
+#  specific date is a holiday as fast and flexible as possible.
+#
+#  Authors: Vacanza Team and individual contributors (see CONTRIBUTORS file)
+#           dr-prodigy <dr.prodigy.github@gmail.com> (c) 2017-2023
+#           ryanss <ryanssdev@icloud.com> (c) 2014-2017
+#  Website: https://github.com/vacanza/holidays
+#  License: MIT (see LICENSE file)
+
+from gettext import gettext as tr
+
+from holidays.calendars import _CustomIslamicHolidays
+from holidays.calendars.gregorian import APR
+from holidays.groups import InternationalHolidays, IslamicHolidays
+from holidays.holiday_base import HolidayBase
+
+
+class Tajikistan(HolidayBase, InternationalHolidays, IslamicHolidays):
+    """Tajikistan holidays.
+
+    References:
+        * <https://en.wikipedia.org/wiki/Public_holidays_in_Tajikistan>
+        * [Law #3 from Nov 3, 1995](https://web.archive.org/web/20250823162418/http://www.portali-huquqi.tj/publicadliya/view_qonunhoview.php?showdetail=&asosi_id=155&language=tj)
+        * [Law #753 from Aug 2, 2011](https://web.archive.org/web/20250823162543/http://www.portali-huquqi.tj/publicadliya/view_qonunhoview.php?showdetail=&asosi_id=12997&language=tj)
+        * [Independence Day](https://web.archive.org/web/20250324203223/https://tnu.tj/index.php/en/9-september-independence-day-of-the-republic-of-tajikistan/)
+    """
+
+    country = "TJ"
+    default_language = "tg"
+    # %s (estimated).
+    estimated_label = tr("%s (таҳминан)")
+    start_year = 1992
+    supported_languages = ("en_US", "ru", "tg")
+
+    def __init__(self, *args, islamic_show_estimated: bool = True, **kwargs):
+        """
+        Args:
+            islamic_show_estimated:
+                Whether to add "estimated" label to Islamic holidays name
+                if holiday date is estimated.
+        """
+        InternationalHolidays.__init__(self)
+        IslamicHolidays.__init__(
+            self, cls=TajikistanIslamicHolidays, show_estimated=islamic_show_estimated
+        )
+        super().__init__(*args, **kwargs)
+
+    def _populate_public_holidays(self):
+        # New Year's Day.
+        self._add_new_years_day(tr("Соли Нав"))
+
+        # Mother's Day.
+        self._add_holiday_mar_8(tr("Рӯзи Модар"))
+
+        # International Nowruz Day.
+        name = tr("Иди байналмилалии Наврӯз")
+        self._add_holiday_mar_21(name)
+        self._add_holiday_mar_22(name)
```

Comment:
Just my 2-cents for later on but I think Nowruz Day logic should be moved to `InternationalHolidays` (Albania and Kazakhstan did observe it on March 22nd AFAIK) - there's an existing `_add_nowruz_day` for `PersianCalendar` used by Afghanistan and Iran so that's something to keep in mind for naming.

Reply:
That's a good idea. But which date should we use as the main date?

Reply:
But if we make March 20 the first day (Azerbaijan and Turkmenistan), then Albania and Kazakhstan will start on ..._day_three. That doesn't look very good...